### ISSUE TITLE
False positive in rule `closure_params_parantheses` with `self` argument name

### ISSUE THREAD

**Issue by Dschee** [realm/SwiftLint#2640]
Using SwiftLint 0.30.1 I get the following false positive:

```swift
// setup view model bindings
viewModel?.profileImage.didSet(weak: self) { (self, profileImage) in // << Don't use parantheses around non-typed parameters in a closure.
    if let profileImage = profileImage {
        self.profileImageView.image = profileImage
    }
    else {
        self.profileImageView.image = UIImage(named: "avatar_100pt")
    }
}
```

When removing the parantheses though as suggested, the code doesn't build anymore.

The underlying implementation looks like this:
```
import Foundation

/// A binding wrapper to track changes on constant properties.
class ObservableProperty<ValueType> {
    // MARK: - Stored Instance Properties
    private var didSetClosures: [(ValueType) -> Void] = []

    private(set) var value: ValueType {
        didSet {
            didSetClosures.forEach { $0(value) }
        }
    }

    // MARK: - Initializers
    init(_ value: ValueType) {
        self.value = value
    }

    // MARK: - Instance Methods
    /// Will be called after the wrapped value is changed.
    /// `$0` is the passed parameter (usually `self`), `$1` is the new value.
    func didSet<WeakObject: AnyObject>(weak object: WeakObject, _ closure: @escaping (WeakObject, ValueType) -> Void) {
        let weakClosure: (ValueType) -> Void = { [weak object] value in
            guard let object = object else { return }
            closure(object, value)
        }

        didSetClosures.append(weakClosure)
        weakClosure(value)
    }

    /// Use to change the value wrapped as a variable.
    func set(_ newValue: ValueType) {
        value = newValue
    }
}
```

This is an implementation of the idea also implemented in https://github.com/dreymonde/Delegated and should therefore be warning-free. My best guess is that there's some kind of exception logic going on in Swift for `self` argument names, but they are allowed since Swift 4 so this should be fixed.

**Comment by Dschee** [comment]
Ahh, forget about this one, it was a custom rule:

```yaml
  closure_params_parantheses:
    included: ".*.swift"
    regex: '\{\s*\([^):]+\)\s*in'
    name: "Unnecessary Closure Params Parantheses"
    message: "Don't use parantheses around non-typed parameters in a closure."
    severity: warning
```

The fix was simple:

```yaml
  closure_params_parantheses:
    included: ".*.swift"
    regex: '\{\s*\((?!self)[^):]+\)\s*in'
    name: "Unnecessary Closure Params Parantheses"
    message: "Don't use parantheses around non-typed parameters in a closure."
    severity: warning
```
